### PR TITLE
Auto release new minor versions

### PR DESCRIPTION
**Pull request by zen0bit**

### CODE REVIEW

**Comment by zen0bit** [comment]
Major version bump should be done manually

**Comment by zen0bit** [comment]
[Preview](https://github.com/oSoWoSo/easybashgui/releases)

**Comment by mckaygerhard** [comment]
:+1:  great! hey, the trigger is? i mean when this "auto" release tag will be deploy? witch action does the triggering ?

**Comment by mckaygerhard** [comment]
i mean you put 

```
on: 
  push:
    branches:
      - master
```

this has a problem, on any commit+push we will have a release no matter is there is a test failing or not!

**Comment by vaisarger** [comment]
I agree with @mckaygerhard ...

"Automated" release naming would be done even in case there is a test failing, or anything else failing, that is, e.g., I *don't want* a new release (at least, I don't want it *yet* ).

**Comment by zen0bit** [comment]
> 👍 great! hey, the trigger is? i mean when this "auto" release tag will be deploy? witch action does the triggering ?

trigger could be whatever you want, could run only after success test.. (change needed)

Or bind to tag (major minor patch no-release)

automated releases could be named like you want (marked as you want pre-release, latest etc)

> so your proposal maybe is : have a stable branch for release and a develop branch ?

That is good practice

**Comment by vaisarger** [comment]
So, if I've got it properly, when I type in my terminal:
`git push github master `

then Github would create, say, the version 12.0.7 of EBG, automatically?

Am I correct ?

**Comment by zen0bit** [comment]
12.1.0

**Comment by vaisarger** [comment]
So, my friend... We would have only 9 possible changes before major version change...

This project is not so "live & working", anyway I see this as a limitation...

@mckaygerhard  what do you think about?

**Comment by zen0bit** [comment]
12.99.0

**Comment by zen0bit** [comment]
its scripting anything isn't in stones (i hope)
anything is impossible , you just need figured out how could be changed...

only boundary is your imagination
right?

**Comment by vaisarger** [comment]
> 12.99.0

Oh ! right ! :P

> Or you can use bash in github action for anything you want..
Namely, you are saying that, e.g. for a potential 12.0.7, I still could do pushing by a new tag from my terminal ?

> anything is possible, you just need figured out how could be changed...
I absolutely agree ! you know... I'm just trying to realize because I'm not at all Github expert: only for this reason!

**Comment by zen0bit** [comment]
> So, my friend... We would have only 9 possible changes before major version change...

you are not! we are working with computers which can oprate with anything we will give them. (not only 1-9) 
image


> This project is not so "live & working", anyway I see this as a limitation...
> 
> @mckaygerhard what do you think about?

**Comment by vaisarger** [comment]
I'm seriously considering it... 

With this auto-versioning... Is still possible changing version manually by tag, from terminal?

**Comment by zen0bit** [comment]
> > 12.99.0
> 
> Oh ! right ! :P
> 
> > Or you can use bash in github action for anything you want..
> 
> Namely, you are saying that, e.g. for a potential 12.0.7, I still could do pushing by a new tag from my terminal ?

sure
 (not yet get used to githubcli) except pushing/pulling everything done on github webui.

or can be triggered only manually. so one click for you with version change (which one you choose) with complet changelog in few seconds

or better don't ask what you can, tell what you want...

> > anything is possible, you just need figured out how could be changed...

anything is possible, you just need figured out how could be changed...

> I absolutely agree ! you know... I'm just trying to realize because I'm not at all Github expert: only for this reason!

Me to, just bit more experienced, since working against actions often 😃

**Comment by vaisarger** [comment]
I hope my question will not mind you... (really):

*If I possibly would change my mind about auto-versioning, how could I come back?*

**Comment by zen0bit** [comment]
that minor major patch no0release could be also override with appropriate label in PR name

look on actions READMEs in repo itself...
[action-automatic-release](
https://github.com/marvinpinto/action-automatic-releases)
[release-action](https://github.com/rohailkhaniazi/release-action)
[action-gh-release](https://github.com/softprops/action-gh-release)

> If I possibly would change my mind about auto-versioning, how could I come back?

just delete or change trigger in action to worflow_dispatch only (manual trigger)

**Comment by zen0bit** [comment]
don't ask what you can, tell what you want...

**Comment by vaisarger** [comment]
> just delete yml file

And where will be that file? In same location of LICENSE file ?

**Comment by vaisarger** [comment]
> look at 1 changed file in this PR PS: added some comments inside file to better understand..

Thank you very much, now is more understandable! :P

Still I didn't understand where will go that .yml file in EBG tarball...

**Comment by zen0bit** [comment]
nowhere. its hidden file just for github.

it's why is under.github/

(like github dotfile)

**Comment by mckaygerhard** [comment]
> I'm seriously considering it...
> 
> With this auto-versioning... Is still possible changing version manually by tag, from terminal?

@zen0bit  change the script to use tags.. that solve the problem for @vaisarger  so he can decide when release using tag marks

**Comment by vaisarger** [comment]
> nowhere. its hidden file just for github.

> it's why is under .github/
I don't have that directory... where is ? (sorry )

**Comment by mckaygerhard** [comment]
> > nowhere. its hidden file just for github.
> 
> > it's why is under .github/
> 
> I don't have that directory... where is ? (sorry )

ls -la will show you.. any dir with dot as prefix will be hidden in linux

also if you dont have the changes from him will not have such change and directory! dont worry

such directory `.github/*` is only for github actions.. nothing more.. 

by the way i posted a message before for him! maybe will help to you

**Comment by zen0bit** [comment]
Or you can create release[anyname] branch and if you push changes there, that will trigger new release..

**Comment by zen0bit** [comment]
> > nowhere. its hidden file just for github.
> 
> > it's why is under .github/
> 
> I don't have that directory... where is ? (sorry )

sure, create only if needed

**Comment by zen0bit** [comment]
I added a lot comments etc. to better see how github workflow works..
Now is more a like template and also "Hello world" to see easy use of bash...

File: .github/workflows/auto-release.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    #branches: '**' #'!master' # excludes master
+    #paths-ignore: [ '**/README.md' ]
+  #pull_request:
+    #branches: '**' # matches every branch
+    #paths-ignore: [ '**/README.md' ]
+
+    # Triggers the workflow if matching tag is created
+    tags: [ 'v*.*' ]
```
Comment (mckaygerhard) [inline .github/workflows/auto-release.yml:21]: now we must explain to @vaisarger  that will only trigerr when make a custom tag.. 

@vaisarger  it means no release will be publised .. and only if you tagged will be publised

**Comment by zen0bit** [comment]
Just define what exactly wanna achieve and your current workflow/staff used

**Comment by mckaygerhard** [comment]
> Just define what exactly wanna achieve and your current workflow/staff used

yes looks good to me!

**Comment by vaisarger** [comment]
Sorry guys, we Open Source developers have a (tough) life also... :P
I would ask you to confirm two things about this PR:

1) If merged, a .github dir will appear in my local project directory?

2) with last change, only if I set a new tag (typing in terminal e.g.: `git tag -a 13.0 -m "Brand new EBG version" ` ), github will trigger a new version... But, typing only: `git push github master` then *no* new version will be created?

Thank you for your patience...

**Comment by zen0bit** [comment]
> 1) If merged, a .github dir will appear in my local project directory?

yes

> 2) with last change, only if I set a new tag (typing in terminal e.g.: `git tag -a 13.0 -m "Brand new EBG version" ` ), github will trigger a new version...

On line 21 tag is set to tags: [ 'v*.*' ]

> But, typing only: `git push github master` then *no* new version will be created?

yes

**Comment by vaisarger** [comment]
Ok. Now...

I usually use these commands to add and push:
`git tag -a "13.0.0" -m "Brand new EBG version" && git push --tags github master`

...

After this PR, what will the list of commands look like?
Maybe this:
`git tag -a "v.13.0.0" -m "Brand new EBG version" && git push master v.13.0.0`

???

**Comment by mckaygerhard** [comment]
> I usually use these commands to add and push: `git tag -a "13.0.0" -m "Brand new EBG version" && git push --tags github master`
> ...
> After this PR, what will the list of commands look like? Maybe this: `git tag -a "v.13.0.0" -m "Brand new EBG version" && git push master v.13.0.0`

umm now makes sense to me your questions. . @zen0bit  needs a letter like "v" to discern if the tag is to make a release, but you do not use letters and use only numbers.. umm i have some ideas to improve your PR @zen0bit  but i will check it in deep only this weekend..

**Comment by mckaygerhard** [comment]
@vaisarger i tested and if you want to be release only on tags we can change the behaviour but liked the idea!

**Comment by zen0bit** [comment]
look at second PR and again at my repo to what happened...

**Comment by zen0bit** [comment]
we can closed this I think...

**Comment by mckaygerhard** [comment]
seems you works so much @zen0bit  yeah. close this and let @vaisarger  check #36 that it follows the schema number as X.Y.Z for tags releases

**Comment by zen0bit** [comment]
Completed

**Pull request closed without merging**